### PR TITLE
Trigger did_buffer_overflow if multi-image neighborlist needs to search more shifts

### PR DESCRIPTION
**Pull request by teddykoker**
Currently, if the box shrinks during simulation (e.g. structure relaxation) the resulting neighbor list can be incorrect if more cell repeats are needed to account for the smaller cell size. This fix, sets `did_buffer_overflow=True` which should prompt the user to rebuild the neighborlist (along with the new cell/repeats) in order to get the correct edges.

## Checklist

- [X] Tests added/updated and passing
- [X] Appropriate labels added to PR

### CODE REVIEW

File: jax_md/custom_partition.py [hunk 1]
```diff
@@ -1035,24 +1038,36 @@ def neighbor_list_fn(
...
+        did_buffer_overflow=(
+          updated.did_buffer_overflow | shift_range_overflow
+        ),
+      )
+
     # If box= was provided the geometry has changed; always rebuild.
     # The displacement threshold only applies when the box is unchanged.
     if use_threshold and 'box' not in kwargs:
       return jax.lax.cond(
         check_needs_rebuild(position, neighbors.reference_position),
```
Comment (abhijeetgangan) [inline jax_md/custom_partition.py]: We should also include reference box?
